### PR TITLE
refactor: error checks and nil pointer check

### PR DESCRIPTION
**Pull request by abhimanyu003**
**What this PR does / why we need it**:

* Removes not-required error checks.
* Nil pointer update for switch case.
* Test cases update

### CODE REVIEW

File: operator/apis/mlops/v1alpha1/experiment_types.go [hunk 1]
```diff
@@ -154,7 +154,7 @@ func (es *ExperimentStatus) IsConditionReady(t apis.ConditionType) bool {
 func (es *ExperimentStatus) SetCondition(conditionType apis.ConditionType, condition *apis.Condition) {
 	switch {
 	case condition == nil:
-		experimentConditionSet.Manage(es).MarkUnknown(conditionType, "", "")
+		return
```
Comment (sakoush) [inline operator/apis/mlops/v1alpha1/experiment_types.go:157]: could you look for all/similar occurrences for this `SetCondition` helper and follow the same pattern you have suggested here?

**Review by sakoush — COMMENTED** [review]
LGTM. I think that there are still other occurrences where this  might happen and need fixing, so perhaps search for `SetCondition` to get all cases.

File: operator/apis/mlops/v1alpha1/experiment_types.go (hunk 1 above)
Comment (abhimanyu003) [reply operator/apis/mlops/v1alpha1/experiment_types.go:157]: I have updated all the cases to have similar pattern  for `SetCondition`

**Review by sakoush — APPROVED** [review]
LGTM

**Pull request merged**